tests/reserve_tests.cpp

```diff
@@ -0,0 +1,192 @@
+#include "account_tester.hpp"
+
+const static name reserve_account_name = N(gxc.reserve);
+
+class gxc_reserve_tester : public gxc_account_tester {
+public:
+
+   gxc_reserve_tester() {
+      create_accounts({ reserve_account_name, N(gamex) });
+      produce_blocks(1);
+
+      _set_code(reserve_account_name, contracts::reserve_wasm());
+      _set_abi(reserve_account_name, contracts::reserve_abi().data());
+
+      auto accnt = control->db().get<account_object, by_name>(reserve_account_name);
+      abi_def abi;
+      BOOST_REQUIRE_EQUAL(abi_serializer::to_abi(accnt.abi, abi), true);
+      abi_ser[reserve_account_name].set_abi(abi, abi_serializer_max_time);
+   }
+
+   action_result setminamount(extended_asset value) {
+      return PUSH_ACTION(reserve_account_name, reserve_account_name, (value));
+   }
+
+   action_result mint(extended_asset derivative, extended_asset underlying, std::vector<option> opts ={}) {
+      return PUSH_ACTION(reserve_account_name, basename(derivative.contract), (derivative)(underlying)(opts));
+   }
+
+   action_result prepay(account_name creator, account_name name, extended_asset value) {
+      return PUSH_ACTION(reserve_account_name, creator, (creator)(name)(value));
+   }
+
+   action_result claim(account_name owner, extended_asset value) {
+      return PUSH_ACTION(reserve_account_name, owner, (owner)(value));
+   }
+
+};
+
+BOOST_AUTO_TEST_SUITE(gxc_reserve_tests)
+
+BOOST_FIXTURE_TEST_CASE(setminamount_tests, gxc_reserve_tester) try {
+
+   gxc_token_tester::mint(EA("10000.0000 GXC@gxc"), false);
+   produce_blocks(1);
+
+   BOOST_REQUIRE_EQUAL(wasm_assert_msg("min amount must not be negative"),
+         setminamount(EA("-1.0000 GXC@gxc"))
+   );
+
+   BOOST_REQUIRE_EQUAL(success(),
+         setminamount(EA("1.0000 GXC@gxc"))
+   );
+
+   gxc_token_tester::mint(EA("10000.0000 HOBL@nosys"), false);
+   BOOST_REQUIRE_EQUAL(wasm_assert_msg("prepaid asset should be system token"),
+         setminamount(EA("1.0000 HOBL@nosys"))
+   );
+   produce_blocks(1);
+
+} FC_LOG_AND_RETHROW()
+
+BOOST_FIXTURE_TEST_CASE(reserve_mint_tests, gxc_reserve_tester) try {
+
+   set_authority(token_account_name, "active",
+         authority(1,
+            vector<key_weight>{{get_private_key("gxc.token", "active").get_public_key(), 1}},
+            vector<permission_level_weight>{{{N(gxc), config::active_name}, 1}, {{N(gxc.reserve), config::eosio_code_name}, 1}, {{N(gxc.token), config::eosio_code_name}, 1}}, {}),
```

Comment:
Unlike the permission of mainnet token account, `gxc@active` seems not essential here, can you check?